src/query/src/sql.rs

```diff
@@ -227,6 +241,97 @@ fn describe_column_semantic_types(
     ))
 }
 
+fn show_create_column_name(table_name: Vec<String>) -> VectorRef {
+    Arc::new(StringVector::from(table_name))
+}
+
+fn show_create_table_column_name(table_info: TableInfoRef) -> VectorRef {
+    let mut create_sql_column: Vec<String> = Vec::new();
+
+    let columns_schemas = table_info.meta.schema.column_schemas();
+    let primary_key_indices = &table_info.meta.primary_key_indices;
+    let table_name = &table_info.name;
+
+    create_sql_column.push(format!("CREATE TABLE {} ( ", table_name));
+    for (index, cs) in columns_schemas.iter().enumerate() {
+        create_sql_column.push(cs.name.clone());
+        create_sql_column.push(cs.data_type.name().to_string());
+        let default_val = if cs.is_nullable() {
+            "null".to_string()
+        } else if cs.default_constraint().is_none() {
+            "".to_string()
+        } else {
+            format!(
+                "default {}",
+                cs.default_constraint()
+                    .map_or(String::from(""), |dc| dc.to_string())
+            )
+        };
+        create_sql_column.push(default_val);
+        if index != columns_schemas.len() - 1 {
+            create_sql_column.push(",".to_string());
+        }
+    }
+
+    let mut create_sql_index: Vec<String> = Vec::new();
+    for (index, cs) in columns_schemas.iter().enumerate() {
+        if primary_key_indices.contains(&index) {
+            create_sql_index.push(format!("primary key({})", cs.name));
+            create_sql_index.push(",".to_string());
+        } else if cs.is_time_index() {
+            create_sql_index.push(format!("time index({})", cs.name));
+            create_sql_index.push(",".to_string());
+        } else {
+        };
+    }
+
+    if !create_sql_index.is_empty() {
+        create_sql_column.push(",".to_string());
+        create_sql_index.remove(create_sql_index.len() - 1);
+        create_sql_index.push(")".to_string());
+    }
+
+    create_sql_index.push(format!(
+        "engine={} with(regions={});",
+        table_info.meta.engine,
+        table_info.meta.region_numbers.len()
+    ));
+
+    let create_sql_column_value = create_sql_column.join(" ");
+    let create_sql_column_index = create_sql_index.join(" ");
+    Arc::new(StringVector::from(vec![format!(
+        "{}{}",
+        create_sql_column_value, create_sql_column_index
+    )]))
+}
+
+pub fn show_create_table(
+    stmt: ShowCreateTable,
+    catalog_manager: CatalogManagerRef,
+) -> Result<Output> {
+    let catalog = stmt.catalog_name.as_str();
+    let schema = stmt.schema_name.as_str();
+    let schema = catalog_manager
+        .schema(catalog, schema)
+        .context(error::CatalogSnafu)?
+        .context(error::SchemaNotFoundSnafu { schema })?;
+    let table = schema
+        .table(&stmt.table_name)
+        .context(error::CatalogSnafu)?
+        .context(error::TableNotFoundSnafu {
+            table: &stmt.table_name,
```

Comment:
It would be more clear to also include "catalog name" and "schema name" in the table. They are usually connected with `.` like `format!("{}.{}.{}", catalog_name, schema_name, table_name)`

Reply:
I'm not sure what it means. Do you want me to show `catalog_name, schema_name, table_name`？

Reply:
Yes, like here https://github.com/GreptimeTeam/greptimedb/blob/75ce2568498cf0f88d4844c77da55948a821b1d9/src/catalog/src/local/manager.rs#L257-L262